operator-framework/src/test/java/io/javaoperatorsdk/operator/sample/primarytosecondaydependent/PrimaryToSecondaryDependentReconciler.java

```diff
@@ -52,6 +52,12 @@ public int getNumberOfExecutions() {
     return numberOfExecutions.get();
   }
 
+  /**
+   * Here just simply creating an Event Source, and setting it for the Dependent Resource. Since it
+   * is not possible to do this setup within the bounds of the KubernetesDependentResource API.
+   * However, this is quite a corner case; might be covered more out of the box in the future if
```

Comment:
Yes, I think we should provide a way for dependents to initialize their event sources if needed in the future.

Reply:
So this is doable also now just not in an elegant way, and did not want to confuse users. One way it would be just to override this method: 

https://github.com/java-operator-sdk/java-operator-sdk/blob/81f7e30ced5bbfe097ca9482e5c7b760ed883320/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/DependentResource.java#L47-L50

The SecondaryToPrimary is supported explicitly, but would not be useful to support the PrimaryToSecondary in same way, since for that (as in the example) also access to the context is needed.

We can think about this in the future more, how to cover also this case more elegantly in KubernetesDependentResource, however it's kinda rare case as mentioned. Since not sure if users might want to use dependent resources for these read-only cases.

Reply:
I would use dependent resources everywhere I can, in a declarative way, so I'm all in favor of making everything possible declaratively (as long as we can do it in an elegant way)… 😄

Reply:
The thing is that in this cases this is a ready only resource, not reallly reconciled, and that is what dependent resources primary does, so at least it is smelly here a little bit. 

Tbh I'm not big fan of annotations, but in this would be good to have possibility to do event sources by annotation as we prototyped in the past - to be consistent at least.